### PR TITLE
updated specs for exporting and importing stories

### LINKED ISSUE #3: Support for exporting and importing stories as JSON
As a developer, I would like to import and export stories in JSON format.

### PR DESCRIPTION
**Pull request by platonovsimeon**
resolves #3

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -1469,6 +1390,87 @@ paths:
             application/json;charset=utf-8:
               schema:
                 $ref: '#/components/schemas/ErrorResponse'
+  /storyImport/:
+    post:
+        operationId: importStory
+        summary: Import a story
+        description: Import story
+        tags:
+          - ExporttedStory
```
Comment (anttileppa) [inline swagger.yaml]: StoryExport

File: swagger.yaml [hunk 2]
```diff
@@ -1469,6 +1390,87 @@ paths:
...
+      parameters:
+        - in: path
+          name: storyId
+          schema:
+            type: string
+            format: uuid
+          required: true
+          description: "Id of the story to be exported"
+      tags:
+        - Exportted Story
```
Comment (anttileppa) [inline swagger.yaml]: StoryExport

File: swagger.yaml [hunk 3]
```diff
@@ -1469,6 +1390,87 @@ paths:
...
+        requestBody:
+          content:
+            application/json;charset=utf-8:
+              schema:
+                $ref: '#/components/schemas/ExporttedStory'
+          description: Payload
+          required: true
+        responses:
+          '200':
+            description: Importted story
```
Comment (anttileppa) [inline swagger.yaml]: Imported

File: swagger.yaml [hunk 4]
```diff
@@ -1469,6 +1390,87 @@ paths:
...
+        operationId: importStory
+        summary: Import a story
+        description: Import story
+        tags:
+          - ExporttedStory
+        requestBody:
+          content:
+            application/json;charset=utf-8:
+              schema:
+                $ref: '#/components/schemas/ExporttedStory'
```
Comment (anttileppa) [inline swagger.yaml]: exported

File: swagger.yaml [hunk 5]
```diff
@@ -1469,6 +1390,87 @@ paths:
...
+          schema:
+            type: string
+            format: uuid
+          required: true
+          description: "Id of the story to be exported"
+      tags:
+        - Exportted Story
+      responses:
+        '200':
+          description: Exportted story
```
Comment (anttileppa) [inline swagger.yaml]: exported

File: swagger.yaml [hunk 6]
```diff
@@ -1469,6 +1390,87 @@ paths:
...
+          content:
+            application/json;charset=utf-8:
+              schema:
+                $ref: '#/components/schemas/ExporttedStory'
```
Comment (anttileppa) [inline swagger.yaml]: exported

File: swagger.yaml [hunk 7]
```diff
@@ -1513,7 +1515,7 @@ components:
           type: string
           format: date-time
           readOnly: true
-    StoryJson:
+    ExporttedStory:
       type: object
       properties:
         id:
```
Comment (anttileppa) [inline swagger.yaml]: remove id property

Comment (anttileppa) [reply swagger.yaml]: Remove readOnly: true properties

File: swagger.yaml [hunk 8]
```diff
@@ -1513,7 +1515,7 @@ components:
           type: string
           format: date-time
           readOnly: true
-    StoryJson:
```
Comment (anttileppa) [inline swagger.yaml:1516]: ExportedStory

File: swagger.yaml (hunk 7 above)
Comment (anttileppa) [reply swagger.yaml]: Move Knots to ExportedStoryKnot, Intents to ExportedStoryIntent, etc..

Comment (anttileppa) [reply swagger.yaml]: Remove variable storyId, remove variable id

Comment (anttileppa) [reply swagger.yaml]: Remove script id

Comment (anttileppa) [reply swagger.yaml]: Move training materials to their own export object

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 1 above)
Comment (platonovsimeon) [reply swagger.yaml]: Changed

File: swagger.yaml (hunk 2 above)
Comment (platonovsimeon) [reply swagger.yaml]: Changed

File: swagger.yaml (hunk 3 above)
Comment (platonovsimeon) [reply swagger.yaml]: changed

File: swagger.yaml (hunk 4 above)
Comment (platonovsimeon) [reply swagger.yaml]: changed

File: swagger.yaml (hunk 5 above)
Comment (platonovsimeon) [reply swagger.yaml]: changed

File: swagger.yaml (hunk 6 above)
Comment (platonovsimeon) [reply swagger.yaml]: changed

File: swagger.yaml (hunk 8 above)
Comment (platonovsimeon) [reply swagger.yaml:1516]: Resolved

File: swagger.yaml (hunk 7 above)
Comment (platonovsimeon) [reply swagger.yaml]: Resolved

File: swagger.yaml [hunk 9]
```diff
@@ -1513,139 +1515,123 @@ components:
...
+          targetKnotId:
+            description: target knot id id.
+            type: string
+            format: uuid
+          global:
+            type: boolean
+          trainingMaterials:
+            type: array
+            items:
+              $ref: '#/components/schemas/ExportedStoryTrainingMaterial'
```
Comment (anttileppa) [inline swagger.yaml]: this should be array of uuid strings

File: swagger.yaml [hunk 10]
```diff
@@ -1513,139 +1515,123 @@ components:
...
+            description: source knot id id.
+            type: string
+            format: uuid
+          targetKnotId:
+            description: target knot id id.
+            type: string
+            format: uuid
+          global:
+            type: boolean
+          trainingMaterials:
```
Comment (anttileppa) [inline swagger.yaml]: training material ids

File: swagger.yaml [hunk 11]
```diff
@@ -1513,139 +1515,123 @@ components:
...
+            type: array
+            items:
+              $ref: '#/components/schemas/ExportedStoryTrainingMaterial'
+    ExportedStoryTrainingMaterial:
+      type:  object
+      properties:
+        id:
+          description: intent id.
```
Comment (anttileppa) [inline swagger.yaml]: training material id

File: swagger.yaml [hunk 12]
```diff
@@ -1513,139 +1515,123 @@ components:
           type: string
           format: date-time
           readOnly: true
-    StoryJson:
+    ExportedStory:
```
Comment (anttileppa) [inline swagger.yaml:1518]: Add training materials array

Comment (anttileppa) [inline swagger.yaml:1518]: Add required to all export properties

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: swagger.yaml [hunk 13]
```diff
@@ -1513,139 +1515,155 @@ components:
...
       type: object
+      required:
+        - name
```
Comment (anttileppa) [inline swagger.yaml:1521]: add intents, knots, etc into required list

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Pull request merged**